packages/typespec-test/test/anomalyDetector/generated/typespec-ts/package.json

```diff
@@ -1,109 +1,111 @@
-    "rollup-plugin-sourcemaps": "^0.6.3",
-    "uglify-js": "^3.4.9",
+    "tshy": "1.11.1",
+    "@azure/core-util": "^1.0.0",
     "@azure-tools/test-credential": "^1.0.0",
     "@azure/identity": "^4.0.1",
     "@azure-tools/test-recorder": "^3.0.0",
-    "mocha": "^10.0.0",
-    "esm": "^3.2.18",
-    "@types/mocha": "^10.0.0",
-    "cross-env": "^7.0.2",
-    "@types/chai": "^4.2.8",
-    "chai": "^4.2.0",
-    "karma-chrome-launcher": "^3.0.0",
-    "karma-coverage": "^2.0.0",
-    "karma-env-preprocessor": "^0.1.1",
-    "karma-firefox-launcher": "^2.1.2",
-    "karma-junit-reporter": "^2.0.1",
-    "karma-mocha-reporter": "^2.2.5",
-    "karma-mocha": "^2.0.1",
-    "karma-source-map-support": "~1.4.0",
-    "karma-sourcemap-loader": "^0.4.0",
-    "karma": "^6.2.0",
-    "c8": "^8.0.0",
-    "ts-node": "^10.0.0"
+    "@vitest/browser": "^1.3.1",
+    "@vitest/coverage-istanbul": "^1.3.1",
+    "playwright": "^1.41.2",
+    "vitest": "^1.3.1"
+  },
+  "scripts": {
+    "clean": "rimraf --glob dist dist-browser dist-esm test-dist temp types *.tgz *.log",
+    "extract-api": "rimraf review && mkdirp ./review && api-extractor run --local",
+    "pack": "npm pack 2>&1",
+    "lint": "eslint package.json api-extractor.json src  --ext .ts",
+    "lint:fix": "eslint package.json api-extractor.json src  --ext .ts --fix --fix-type [problem,suggestion]",
+    "unit-test": "npm run unit-test:node && npm run unit-test:browser",
+    "unit-test:browser": "vitest -c vitest.browser.config.ts",
+    "unit-test:node": "vitest -c vitest.config.ts",
+    "integration-test": "npm run integration-test:node && npm run integration-test:browser",
+    "integration-test:browser": "echo skipped",
+    "integration-test:node": "echo skipped",
+    "check-format": "prettier --list-different --config ../../../.prettierrc.json --ignore-path ../../../.prettierignore \"src/**/*.ts\" \"*.{js,json}\"  \"test/**/*.ts\" \"samples-dev/**/*.ts\"",
+    "format": "prettier --write --config ../../../.prettierrc.json --ignore-path ../../../.prettierignore \"src/**/*.ts\" \"*.{js,json}\"  \"test/**/*.ts\" \"samples-dev/**/*.ts\"",
+    "test:browser": "npm run clean && npm run build:test && npm run unit-test:browser && npm run integration-test:browser",
+    "test:node": "npm run clean && tshy && npm run unit-test:node && npm run integration-test:node",
+    "test": "npm run clean && tshy && npm run unit-test:node && npm run unit-test:browser && npm run integration-test",
+    "build": "npm run clean && tshy && npm run extract-api"
   },
-  "browser": {
-    "./dist-esm/test/public/utils/env.js": "./dist-esm/test/public/utils/env.browser.js"
+  "exports": {
+    "./package.json": "./package.json",
+    ".": {
+      "browser": {
+        "types": "./dist/browser/index.d.ts",
+        "default": "./dist/browser/index.js"
+      },
+      "react-native": {
+        "types": "./dist/react-native/index.d.ts",
+        "default": "./dist/react-native/index.js"
```

Comment:
interesting, should the cjs code be the same as browser one after building?

another detail to confirm is that the replacement logic will be still working, right?

https://github.com/Azure/autorest.typescript/blob/main/packages/autorest.typescript/test/smoke/generated/anomaly-detector-rest/package.json#L103